chapter05/mssw/exercise2.pl

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/perl
+use v5.10;
+use strict;
+use warnings;
+use Data::Dumper;
+
+my %total_bytes;
+
+my $filename = 'chapter05/mssw/coconet.dat';
+
+# my $filename = 'chapter05/mssw/coconet-sample.dat';  # smaller sample
+
+open( my $fh, '<:encoding(UTF-8)', $filename )
+  or die "could not open file '$filename' $!";
+
+while ( my $row = <$fh> ) {
+
+    # check if the line starts with a '#' and skip it
+    if ( $row =~ /^\s*#/ ) {
+        next;
+    }
+    my ( $source, $destination, $bytes ) = split /\s+/, $row;
+    $total_bytes{$source}{$destination} += $bytes;
+}
+
+# Lets create a complex hash! \o/
+#  %report = (
+#   $source => {
+#     destinations = {
+#       $destination => $bytes,
+#       ...
+#     },
+#     total = $total_bytes,
+#   },
+#   ...
+# );
+my %report;
+foreach my $source ( sort keys %total_bytes ) {
+    for my $destination ( sort keys %{ $total_bytes{$source} } ) {
+        $report{$source}->{destinations}{$destination} +=
+          $total_bytes{$source}{$destination};
+        $report{$source}->{total} += $total_bytes{$source}{$destination};
```

Comment:
I thought you need at least one arrow to get the party started? seeing as `%report` isn't a reference